pkg/graveler/ref/manager.go

```diff
@@ -163,16 +165,61 @@ func (m *KVManager) ListRepositories(ctx context.Context) (graveler.RepositoryIt
 	return NewKVRepositoryIterator(ctx, &m.kvStore)
 }
 
+func (m *KVManager) updateRepoState(ctx context.Context, repo *graveler.RepositoryRecord, state graveler.RepositoryState) error {
+	repo.State = state
+	return m.kvStore.SetMsg(ctx, graveler.RepositoriesPartition(), []byte(graveler.RepoPath(repo.RepositoryID)), graveler.ProtoFromRepo(repo))
+}
+
+func (m *KVManager) deleteRepository(ctx context.Context, repo *graveler.RepositoryRecord) error {
+	// ctx := context.Background() TODO (niro): When running this async create a new context and remove ctx from signature
+	logger := logging.FromContext(ctx)
+	// Scan through repository partition and delete entities, using SecondaryIndex struct since we don't really need the data, just the key
+	// Failures on deleting repository entities don't constitute as repository deletion failure
+	partition := graveler.RepoPartition(repo)
+	it, err := kv.NewPartitionIterator(ctx, m.kvStore.Store, (&kv.SecondaryIndex{}).ProtoReflect().Type(), partition)
+	if err != nil {
```

Comment:
You should exit here.

Reply:
I disagree. The most important part is the deletion of the repository itself. I prefer returning a successful removal of the repository and leaving garbage in the DB than failing the operation. WDYT?

Reply:
I think this is an unexpected error and hence should not be ignored.

Reply:
RepositoryState_IN_DELETION